servers/quarkus-server/src/test/java/org/projectnessie/server/QuarkusTestUtils.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server;
+
+import java.util.Objects;
+
+public final class QuarkusTestUtils {
+
+  private QuarkusTestUtils() {
+    // utillity class
+  }
+
+  public static Integer getQuarkusTestPort() {
+    return Objects.requireNonNull(Integer.getInteger("quarkus.http.test-port"));
+  }
+
+  public static String getNessieApiV1Url() {
+    return String.format("http://localhost:%d/api/v1", getQuarkusTestPort());
```

Comment:
I tend to think that it might be nicer to extract the Nessie URL discovery logic into a JUnit5 extension.... similar to how `@NessieUri` parameters are handled.

Reply:
Basically, we could add something like a `NessieQuarkusTestExtension` to resolve `@NessieUri` parameters based on quarkus-managed system properties instead of `QuarkusTestUtils`.

`AbstractTestQuarkusRest` would declare `@ExtendWith(NessieQuarkusTestExtension.class)` and get the URL injected in a manner similar to `AbstractTestJerseyRest`. WDYT?

Reply:
It it works well, we could even have a `@BeforeEach initApi(@NessieUri uri)` in `AbstractRest` and drop custom calls to `initApi(client)` overrides (except for `TestJerseyRestNaiveClientInMemory`)

Reply:
i've added an extension that will resolve `URI quarkusNessieUri` parameters...

using `@NessieUri` would have been more difficult because it exists in a different module (jaxrs extension which is not realted to quarkus afaict)...
generally i think if we need to support different types like api versions we should overhaul the extensions when this api classes get introduced since then we have the full context and are not playing a guessing game, WDYT?

Reply:
Yes, extensions will have to be adjucted, of course